amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -160,7 +166,7 @@ RecordFlow evictAckedEventAndScheduleNextEvent(RecordsDeliveryAck recordsDeliver
             // Update the triggering flow for post scheduling upstream request.
             flowToBeReturned = recordsRetrievedContext.getRecordFlow();
             // Try scheduling the next event in the queue, if available.
-            if (recordsDeliveryQueue.peek() != null) {
+            if (!isAwaitingFinalAckForCurrentSubscription && recordsDeliveryQueue.peek() != null) {
                 subscriber.onNext(recordsDeliveryQueue.peek().getRecordsRetrieved());
```

Comment:
given we schedule the next event under two different conditions (here and in bufferCurrentEventAndScheduleIfRequired method) , and both required isShuttingDown check, let's move out the schedule call it its own method, with the check. And add comment on why the check is necessary. 

i.e. scheduleNextEvent(event) {if !pendingShutdown; then subscriber.onNext(event)}

Reply:
Ya. makes sense